test/Dialect/FIRRTL/legacy-wiring-errors.mlir

```diff
@@ -59,3 +59,34 @@ firrtl.circuit "Foo" attributes {
       firrtl.skip
   }
 }
+
+// -----
+// Error if attempt to wire incompatible types.
+
+firrtl.circuit "Foo" attributes {
+ rawAnnotations = [
+  {
+    class = "firrtl.passes.wiring.SourceAnnotation",
+    target = "~Foo|Bar>y",
+    pin = "xyz"
+  },
+  {
+    class = "firrtl.passes.wiring.SinkAnnotation",
+    target = "~Foo|Foo>x",
+    pin = "xyz"
+  }
+  ]} {
+  firrtl.module private @Bar() {
+    // expected-error @below {{Wiring Problem source type '!firrtl.bundle<a: uint<1>, b: uint<2>>' does not match sink type '!firrtl.uint<1>'}}
+    %y = firrtl.wire interesting_name : !firrtl.bundle<a: uint<1>, b: uint<2>>
+    %invalid_reset = firrtl.invalidvalue : !firrtl.bundle<a: uint<1>, b: uint<2>>
+    firrtl.strictconnect %y, %invalid_reset : !firrtl.bundle<a: uint<1>, b: uint<2>>
+  }
```

Comment:
Supreme nit:

The invalid value and connect can be removed from the test. It won't pass initialization checking, but keeps the test sharp. Ditto for other tests.

Reply:
Sure thing, thanks!

After doing this, looks like the wire-to-node in-place conversion unexpectedly triggers here.
As-is, the wires get replaced with a node of the source type (since no 'down' ports, only up the hierarchy in these tests).

Will look into how this might best be handled.